vector/src/main/java/im/vector/app/features/link/LinkHandlerActivity.kt

```diff
@@ -45,30 +44,38 @@ class LinkHandlerActivity : VectorBaseActivity<ActivityProgressBinding>() {
     @Inject lateinit var errorFormatter: ErrorFormatter
     @Inject lateinit var permalinkHandler: PermalinkHandler
 
+    override fun getBinding() = ActivityProgressBinding.inflate(layoutInflater)
+
     override fun injectWith(injector: ScreenComponent) {
         injector.inject(this)
     }
 
-    override fun getBinding() = ActivityProgressBinding.inflate(layoutInflater)
-
     override fun initUiAndData() {
-        val uri = intent.data
+        handleIntent()
+    }
 
-        if (uri == null) {
-            // Should not happen
-            Timber.w("Uri is null")
-            finish()
-            return
-        }
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        handleIntent()
+    }
 
-        if (uri.getQueryParameter(LoginConfig.CONFIG_HS_PARAMETER) != null) {
-            handleConfigUrl(uri)
-        } else if (SUPPORTED_HOSTS.contains(uri.host)) {
-            handleSupportedHostUrl(uri)
-        } else {
-            // Other links are not yet handled, but should not come here (manifest configuration error?)
-            toast(R.string.universal_link_malformed)
-            finish()
+    private fun handleIntent() {
+        val uri = intent.data
+        when {
+            uri == null                                                                    -> {
+                // Should not happen
+                Timber.w("Uri is null")
+                finish()
+            }
+            uri.getQueryParameter(LoginConfig.CONFIG_HS_PARAMETER) != null                 -> handleConfigUrl(uri)
+            uri.toString().startsWith(PermalinkService.MATRIX_TO_URL_BASE)                 -> handleSupportedHostUrl()
+            uri.toString().startsWith(PermalinkHandler.MATRIX_TO_CUSTOM_SCHEME_URL_BASE)   -> handleSupportedHostUrl()
+            resources.getStringArray(R.array.permalink_supported_hosts).contains(uri.host) -> handleSupportedHostUrl()
```

Comment:
Don't you think we should consider the case of "clientPermalinkBaseUrl" here too?

Reply:
Yes it makes sense, thanks